skbio/io/fasta.py

```diff
@@ -653,46 +653,49 @@ def _fasta_to_biological_sequence(fh, qual=FileSentinel, seq_num=1):
 
 
 @register_reader('fasta', DNA)
-def _fasta_to_dna_sequence(fh, qual=FileSentinel, seq_num=1):
+def _fasta_to_dna_sequence(fh, qual=FileSentinel, seq_num=1, **kwargs):
     return _get_nth_sequence(
         _fasta_to_generator(fh, qual=qual,
-                            constructor=partial(DNA, validate=False)),
+                            constructor=DNA, **kwargs),
         seq_num)
 
 
 @register_reader('fasta', RNA)
-def _fasta_to_rna_sequence(fh, qual=FileSentinel, seq_num=1):
+def _fasta_to_rna_sequence(fh, qual=FileSentinel, seq_num=1, **kwargs):
     return _get_nth_sequence(
         _fasta_to_generator(fh, qual=qual,
-                            constructor=partial(RNA, validate=False)),
+                            constructor=RNA, **kwargs),
         seq_num)
 
 
 @register_reader('fasta', Protein)
-def _fasta_to_protein_sequence(fh, qual=FileSentinel, seq_num=1):
+def _fasta_to_protein_sequence(fh, qual=FileSentinel, seq_num=1, **kwargs):
     return _get_nth_sequence(
         _fasta_to_generator(fh, qual=qual,
-                            constructor=partial(Protein, validate=False)),
+                            constructor=Protein, **kwargs),
         seq_num)
 
 
 @register_reader('fasta', SequenceCollection)
 def _fasta_to_sequence_collection(fh, qual=FileSentinel,
-                                  constructor=Sequence):
+                                  constructor=Sequence, **kwargs):
     return SequenceCollection(
-        list(_fasta_to_generator(fh, qual=qual, constructor=constructor)))
+        list(_fasta_to_generator(fh, qual=qual, constructor=constructor,
+                                 **kwargs)))
 
 
 @register_reader('fasta', Alignment)
-def _fasta_to_alignment(fh, qual=FileSentinel, constructor=Sequence):
+def _fasta_to_alignment(fh, qual=FileSentinel, constructor=Sequence, **kwargs):
     return Alignment(
-        list(_fasta_to_generator(fh, qual=qual, constructor=constructor)))
+        list(_fasta_to_generator(fh, qual=qual, constructor=constructor,
+                                 **kwargs)))
 
 
 @register_writer('fasta')
 def _generator_to_fasta(obj, fh, qual=FileSentinel,
                         id_whitespace_replacement='_',
-                        description_newline_replacement=' ', max_width=None):
+                        description_newline_replacement=' ', max_width=None,
+                        **kwargs):
```

Comment:
I think this should explicitly be `length` that way it is documented. It doesn't look like kwargs actually get to go anywhere unlike in the `readers` where they get passed to the constructor (where they can fail). Using kwargs here will just result in swallowing any misspellings. `seq.write('filename', id_whitespace_replacment=' ')` won't raise an error and won't do what the user wants.